installer/alpine/Dockerfile

```diff
@@ -1,100 +1,101 @@
 FROM alpine:3.7
 
-ENTRYPOINT /build.sh
+ENTRYPOINT [ "/build.sh" ]
 VOLUME /assets
 
 COPY qemu-aarch64-static /usr/bin/
 
-RUN mkdir -p /etc/apk/cache
-RUN apk add --update --upgrade \
-    alpine-base \
-    curl \
-    docker \
-    jq \
-    mdadm \
-    mkinitfs \
-    musl-utils \
-    openssh \
-    squashfs-tools \
-    tcpdump \
-    ;
-RUN apk cache sync
-RUN mv /etc/apk/cache /cache
-RUN apk add --no-scripts --no-cache --update --upgrade \
-    abuild \
-    alpine-sdk \
-    build-base \
-    busybox-initscripts \
-    coreutils \
-    linux-headers \
-    ;
-
-# remove edge when linux-vanilla is > 4.14.8 and has the nfp driver enabled,
-# likely alpine v3.8
-RUN apk add --no-scripts --no-cache --update --upgrade --repository http://dl-cdn.alpinelinux.org/alpine/edge/main \
-    linux-firmware \
-    ;
-
-RUN adduser -G abuild -g "Alpine Package Builder" -s /bin/ash -D builder && \
+RUN mkdir -p /etc/apk/cache && \
+    apk add --update --upgrade \
+        alpine-base \
+        curl \
+        docker \
+        jq \
+        mdadm \
+        mkinitfs \
+        musl-utils \
+        openssh \
+        squashfs-tools \
+        tcpdump \
+        && \
+    apk add --no-scripts --update --upgrade --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing \
+        kexec-tools \
+        && \
+    apk cache sync && \
+    mv /etc/apk/cache /cache && \
+    apk add --no-scripts --no-cache --update --upgrade \
+        abuild \
+        alpine-sdk \
+        build-base \
+        busybox-initscripts \
+        coreutils \
+        linux-headers \
+        sudo \
+        && \
+    # remove edge when linux-vanilla is > 4.14.8 and has the nfp driver enabled,
+    # likely alpine v3.8
+    apk add --no-scripts --no-cache --update --upgrade --repository http://dl-cdn.alpinelinux.org/alpine/edge/main \
+        linux-firmware \
+	&& \
+    adduser -G abuild -g "Alpine Package Builder" -s /bin/ash -D builder && \
     echo "builder ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers && \
     apk update
 
 USER builder
 WORKDIR /home/builder
 
-# Setup some self-signed keys and grab alpine package tree
-RUN abuild-keygen -a -i -n && \
-    git clone -b master https://github.com/alpinelinux/aports && \
-    cd aports && \
-    git checkout bb52b0d3f60cd75197ef218362f193378dfd869d
-
-# Build customized linux-vanilla
-WORKDIR /home/builder/aports/main/linux-vanilla
-
-RUN sed -i 's/silentoldconfig/olddefconfig/g' APKBUILD && \
-    echo 'CONFIG_KEXEC=y' >> config-vanilla.x86_64 && \
+ENV COMMIT bb52b0d3f60cd75197ef218362f193378dfd869d
+ENV FLAVOR vanilla
+ENV KERNEL 4.14.55
+ENV MAKEFLAGS -j$(($(nproc) * 2))
+# make sure PKGREL is +1 what is in APKGBUILD
+ENV PKGREL 1
+
+RUN true && \
+    # Setup self-signed keys to sign the built packages \
+    abuild-keygen -a -i -n && \
+    # Fetch alpine package tree \
+    curl -fL https://github.com/alpinelinux/aports/archive/$COMMIT.tar.gz | tar -zxf - && \
```

Comment:
do we care about failure scenarios ?

Reply:
More than handling via pipefail? https://github.com/tinkerbell/osie/pull/79/files#diff-b67911656ef5d18c4ae36cb6741b7965R2

Reply:
My question was geared around what error handling strategy we use... as I see this in docker file I understand now.  Withdrawing.